search/search_service/proxy/es_search_proxy.py

```diff
@@ -0,0 +1,316 @@
+
+            if filter.operation == 'OR':
+                filter_queries.append(Q(BOOL_QUERY, should=queries_per_term, minimum_should_match=1))
+
+            elif filter.operation == 'AND':
+                for q in queries_per_term:
+                    filter_queries.append(q) 
+
+            else:
+                raise ValueError(f"Invalid filter operation {filter.operation} for filter {filter_name} with values {filter.values}")
+    
+        return filter_queries
+
+
+    def _build_elasticsearch_query(self, *,
+                                   resource: Resource,
+                                   query_term: str,
+                                   filters: List[Filter]) -> Q:
+
+        term_query = self._build_term_query(resource=resource,
+                                            query_term=query_term)
+        filters = self._build_filters(resource=resource, filters=filters)
+
+        es_query = None
+
+        if filters and term_query:
+            es_query = Q('bool', should=[term_query], filter=filters)
+        elif not filters and term_query:
+            es_query = Q('bool', should=[term_query])
+        elif filters and not term_query:
+            es_query = Q('bool', filter=filters)
+        else:
+            raise ValueError("Invalid search query")
+
+        return es_query
+
+
+    def _format_response(self, page_index: int,
+                         results_per_page: int,
+                         responses: List[Response],
+                         resource_types: List[Resource]) -> dict:
+        resource_types_str = [r.name.lower() for r in resource_types]
+        no_results_for_resource = {
+            "results": [],
+            "total_results": 0
+        }
+        results_per_resource = {resource: no_results_for_resource for resource in resource_types_str}
+
+        for r in responses:
+            if r.success():
+                results_count = r.hits.total.value
+                if results_count > 0:
+                    resource_type = r.hits.hits[0]._type
+                    results = []
+                    for search_result in r.hits.hits:
+                        # mapping gives all the fields in the response
+                        result = {}
+                        fields = self.RESOUCE_TO_MAPPING[Resource[resource_type.upper()]]
+                        for f in fields.keys():
+                            # remove "raw" from mapping value
+                            field = fields.get(f).split('.')[0]
+                            result[f] = search_result._source[field]
+                        result["search_score"] = search_result._score
+                        results.append(result)
+                    # replace empty results with actual results
+                    results_per_resource[resource_type] = {
+                        "results": results,
+                        "total_results": results_count
+                    }
```

Comment:
let's think about how to make this nicer

Reply:
I think this will get way nicer when we have more standard mappings tbh, also having some sort of marshmallow schema for these results